File: web/src/lib/checkpoint-style.ts
import type { CheckpointType } from "./types";

// チェックポイント種別の表示定義。リスト行と地図ピンで共通に使う
// (iOS 側 ios/TripNote/Views/CheckpointStyle.swift と対応)

export const CHECKPOINT_LABELS: Record<CheckpointType, string> = {
  departure: "出発地",
  destination: "到着予定地",
  sightseeing: "観光",
  cafe: "カフェ",
  restaurant: "食事",
  lodging: "宿泊",
  other: "その他",
};

export const CHECKPOINT_ICONS: Record<CheckpointType, string> = {
  departure: "🚩",
  destination: "🏁",
  sightseeing: "🔭",
  cafe: "☕️",
  restaurant: "🍽️",
  lodging: "🛏️",
  other: "📍",
};

// 地図ピン・一覧の点の色(iOS の CheckpointType.tint と同じ値)。
// 暗い地図・パネルの上で読める彩度に寄せてある
export const CHECKPOINT_COLORS: Record<CheckpointType, string> = {
  departure: "#7BD389",
  destination: "#FF7C6B",
  sightseeing: "#F2A03D",
  cafe: "#C99B6A",
  restaurant: "#F274A6",
  lodging: "#8CA0FF",
  other: "#93A0AE",
};

export const TRANSPORT_LABELS: Record<string, string> = {
  car: "車",
  train: "電車",
  walk: "徒歩",
  bicycle: "自転車",
  mixed: "混合",
};

export function transportLabel(transport: string | null): string | null {
  if (!transport) return null;
  return TRANSPORT_LABELS[transport] ?? transport;
}

File: ios/TripNote/Views/CheckpointStyle.swift
import SwiftUI

/// チェックポイント種別の表示定義。リスト行と地図ピンで共通に使う
extension CheckpointType {
    var label: String {
        switch self {
        case .departure: "出発地"
        case .destination: "到着予定地"
        case .sightseeing: "観光"
        case .cafe: "カフェ"
        case .restaurant: "食事"
        case .lodging: "宿泊"
        case .other: "その他"
        }
    }

    var systemImage: String {
        switch self {
        case .departure: "flag.fill"
        case .destination: "flag.checkered"
        case .sightseeing: "binoculars.fill"
        case .cafe: "cup.and.saucer.fill"
        case .restaurant: "fork.knife"
        case .lodging: "bed.double.fill"
        case .other: "mappin"
        }
    }

    /// 暗い地図・パネルの上で読める彩度に寄せた種別色
    /// (Web 側 web/src/lib/checkpoint-style.ts の CHECKPOINT_COLORS と同じ値)
    var tint: Color {
        switch self {
        case .departure: Theme.done
        case .destination: Theme.danger
        case .sightseeing: Color(hex: 0xF2A0_3D)
        case .cafe: Color(hex: 0xC99B_6A)
        case .restaurant: Color(hex: 0xF274_A6)
        case .lodging: Color(hex: 0x8CA0_FF)
        case .other: Theme.muted
        }
    }
}

extension Transport {
    var label: String {
        switch self {
        case .car: "車"
        case .train: "電車"
        case .walk: "徒歩"
        case .bicycle: "自転車"
        case .mixed: "混合"
        }
    }
}
